src/electionguard/api/record_ballots.c

```diff
@@ -201,25 +231,39 @@ bool export_ballots(char *export_path, char *filename_prefix, char **output_file
 {
     bool ok = true;
     char *default_prefix = "electionguard_ballots-";
-    *output_filename = malloc(FILENAME_MAX * sizeof(char));
-    ok = generate_unique_filename(export_path, filename_prefix, default_prefix, *output_filename);   
+    *output_filename = malloc(FILENAME_MAX + 1);
```

Comment:
Similar comment as before. error handling is not clear or documented. `*output_filename` allocation is not handled within the function in case of error.